docs/decisions/0018-expose-all-sass-vars-as-css-vars.rst

```diff
@@ -0,0 +1,75 @@
+0018 CSS Variables
+##################
+
+
+Status
+******
+
+Draft
+
+Context
+*******
+
+.. This section describes the forces at play, including technological, political, social, and project local. These forces are probably in tension, and should be called out as such. The language in this section is value-neutral. It is simply describing facts.
+
+All Open edX MFEs use Bootstrap via Paragon for styling. Each MFE builds its
+own styling which is a combination of common styles from Bootstrap + Paragon,
+on which it addds its own SCSS.
+
+If we want to be able to support a common pre-built core stylesheet that can be
+deployed independently of MFEs, then MFEs cannot use any SCSS variables from
+Paragon + Bootstrap since in that case any changes to the core theme will
+require a rebuild of each MFE.
+
+Decision
+********
+
+We will export all theme variables that an MFE could need as css variables so
+that these can be applyed at runtime rather than build time.
+
+Bootstrap already exports a number of CSS variables (these can be found in
+`this document for Bootstrap 4.x`_), however they don't cover all usecases. For
+instance, usage of colour variables like `primary-400`, or `light-300` are
+abundant. Often they can be used via existing Bootstrap classes, but sometimes
+they can't.
+
+Paragon can create CSS variable exports like: `--pa-blue`, `--pa-primary`,
+`--pa-primary-200`` etc. The reason for prefixing with `--pa` is to isolate
+changes to these names in Bootstrap. For example, in Bootstrap 4.x these are
+named as `--blue, --primary` etc. however in `Bootstrap 5`_, they are prefixed
+with `--bs-blue, --bs-primary`.
+
+Consequences
+************
+
+Wherever we are currently using SASS variables, we can now use CSS variables.
```

Comment:
[clarification] Are these consequences for the consumers of Paragon (e.g., MFEs), or for the component library itself? Or both? It might be worth separating out how each stakeholder might be impacted by CSS variables.

Reply:
Sure, I will do that. 

But roughly, my idea was to leave Paragon's usage of SCSS intact, since it is a shared library, so it can be built and deployed once. I think all of Paragon's SCSS is included with each MFE anyway so right now it doesn't matter as much. 

However, in the future if all of Paragon also uses CSS variables then it too can be built once, and ideally we boil down the compilation to just the variable definitions. The benefit of that will be much quicker compilation of the "theme", and potential near-real-time theme previewing.